### PR TITLE
[Feature]: Add cache to random func in data transform

### PR DESCRIPTION
**Pull request by YuanLiuuuuuu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/transforms/processing.py [hunk 1]
```diff
@@ -1159,6 +1160,7 @@ def _flip(self, results: dict) -> None:
                 results['gt_semantic_seg'],
                 direction=results['flip_direction'])
 
+    @cacheable_method
     def _flip_on_direction(self, results: dict) -> None:
```
Comment (Harold-lkk) [inline mmcv/transforms/processing.py:1164]: The cacheable_method may decorate `_choose_direction` to save `cur_dir`

File: mmcv/transforms/processing.py [hunk 2]
```diff
@@ -1317,6 +1319,7 @@ def _random_sample_ratio(
         scale = int(scale[0] * ratio), int(scale[1] * ratio)
         return scale
 
+    @cacheable_method
```
Comment (Harold-lkk) [inline mmcv/transforms/processing.py:1322]: Maybe to decorate `_random_sample_ratio` and `_random_sample` to save the `scale`

**Comment by Harold-lkk** [comment]
need test

File: mmcv/transforms/processing.py [hunk 3]
```diff
@@ -1336,12 +1338,11 @@ def _random_sample_ratio(
...
+    def _random_scale(self) -> None:
         """Private function to randomly sample an scale according to the type
         of `scale`.
 
-        Args:
-            results (dict): Result dict from :obj:`dataset`.
         Returns:
             dict: One new key 'scale`is added into ``results``,
             which would be used by subsequent pipelines.
```
Comment (mzr1996) [inline mmcv/transforms/processing.py:1348]: Please fix the docstring

File: mmcv/transforms/processing.py [hunk 4]
```diff
@@ -1336,12 +1338,11 @@ def _random_sample_ratio(
...
-    def _random_scale(self, results: dict) -> None:
+    @cacheable_method
+    def _random_scale(self) -> None:
         """Private function to randomly sample an scale according to the type
         of `scale`.
 
-        Args:
-            results (dict): Result dict from :obj:`dataset`.
         Returns:
             dict: One new key 'scale`is added into ``results``,
```
Comment (plyfager) [inline mmcv/transforms/processing.py:1347]: Return a tuple?

**Review by Harold-lkk — APPROVED** [review]

**Pull request closed without merging**